src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -223,27 +224,70 @@ private static IResourceBuilder<ProjectResource> WithProjectDefaults(this IResou
             builder.WithAnnotation(new LaunchProfileAnnotation(launchProfileName));
         }
 
+        var config = GetConfiguration(projectResource);
```

Comment:
I think this code should run even if we've excluded the launch profile right?

Reply:
Ah yes, you're right, good catch. I'll rework this.